xapian-core/include/xapian/cluster.h

```diff
@@ -316,10 +407,30 @@ class XAPIAN_VISIBILITY_DEFAULT RoundRobin {
     ~RoundRobin();
 
     // Implements RoundRobin clustering
-    ClusterSet cluster(MSet &mset, unsigned int k);
+    ClusterSet cluster(MSet &mset, unsigned int num_of_clusters);
 
     // Returns the description of the clusterer being used
     std::string get_description();
 };
-}
+
+/** Kmeans clusterer:
+ *  This clusterer implements the K-Means clustering algorithm
+ */
+class XAPIAN_VISIBILITY_DEFAULT KMeans {
+    std::vector<Point> docs;
+    //std::map<int, docid> doc_list;
+    std::vector<Point> centroids;
+  public:
+    void initialize_centroids(const unsigned int k, ClusterSet &cset);
+
+    void initialize_points(MSetDocumentSource docs, TermListGroup tlg);
+
+    // Implements KMeans clustering
+    ClusterSet cluster(MSet mset, unsigned int k);
```

Comment:
Does `k` here have the match `k` in `initialize_centroids()`? I'd have thought this should 

Also, you really want `KMeans` and `RoundRobin` to be subclasses of the same (abstract) base, so that they can be genuinely interchangeable.